src/test/java/com/google/step/servlets/HomePageServletTest.java

```diff
@@ -97,4 +108,35 @@ public void testDoGet_success() throws Exception {
             expectedDeals, expectedDeals, expectedDeals, expectedDeals);
     JSONAssert.assertEquals(expected, stringWriter.toString(), JSONCompareMode.STRICT);
   }
+
+  @Test
+  public void testDoGet_UserNotLoggedIn() throws Exception {
```

Comment:
Should there be another test for when the user **is** logged in?

Reply:
its testDoGet_success